### ISSUE TITLE
Request code to implement XENet paper

### ISSUE THREAD

**Issue by Tyqfat** [pyg-team/pytorch_geometric#8257]
### 🚀 The feature, motivation and pitch

- Paper: XENet: Using a new graph convolution to accelerate the timeline for protein design on quantum computers
-  This paper introduces XENet, a new graph convolutional network designed to improve the representation of protein environments in sequence design protocols. By paying attention to both incoming and outgoing edge attributes, XENet is able to better model local kinematics problems such as protein design. The paper compares XENet to existing graph convolutions and demonstrates its ability to decrease rotamer sample counts in Rosetta's rotamer substitution protocol by 40% without loss in quality. This allows larger protein design problems to fit onto near-term quantum computers. Additionally, XENet displays an ability to handle deeper architectures than competing convolutions. The authors believe that their work is relevant to the field of protein design and can lead to further advancements in artificial intelligence brought forth by graph neural networks.
- Main Functions: XEnet can take node features and edge features as input, output potential node features and edge features, and can be used for both node classification and edge classification.

### Alternatives

The code solution is:
In TensorFlow 2, We can use the XENetConv class in the Spektral library to implement the Xception convolutional layer.

### Additional context

![image](https://github.com/pyg-team/pytorch_geometric/assets/100210376/c21f27d9-7e51-4ecb-92ba-4321fd529ca3)

**Comment by rusty1s** [comment]
Would definitely like to add to PyG. Do you have interest in contributing this?

**Comment by Tyqfat** [comment]
Absolutely! I am very interested in contributing to this project. As someone who primarily uses PyTorch as my deep learning framework, I find it fascinating. Although I am just getting started with PyG and may need to learn more about it, I am willing to participate in the contribution. Could you please let me know what preparations I need to make before starting?

**Comment by rusty1s** [comment]
No particular preparations needed. I think we can just work on a PR that adds the layer + a test case (similar to how other layers are implemented). Doesn't need to be perfect at the beginning.

**Comment by Tyqfat** [comment]
Thank you very much for your reply, I will give it a try!

**Comment by DarthRevan07** [comment]
@Tyqfat Am i a welcome guest to this party? I have been working in Variational Quantum Algorithms, and Tensor Networks in QC.
Are you currently working on this? Can i try working this out?

**Comment by Tyqfat** [comment]
> @Tyqfat Am i a welcome guest to this party? I have been working in Variational Quantum Algorithms, and Tensor Networks in QC. Are you currently working on this? Can i try working this out?
At present, I am using GNN to solve the unit commitment problem in power system. Because the existing literature uses XENet to solve this problem, I may need to verify whether my strategy is effective based on XENet.

**Comment by minirawat** [comment]
@rusty1s @Tyqfat @DarthRevan07 Can you please review the PR to see if it meets the requirements and if yes, can it be merged? Thanks.

**Comment by thoratsayali517-web** [comment]
I'd like to work on this issue.
